crawl-ref/source/items.cc

```diff
@@ -2843,12 +2843,21 @@ static bool _is_option_autopickup(const item_def &item, bool ignore_force)
     return Options.autopickups[item.base_type];
 }
 
+static int _get_chunk_count() {
+    auto it = find_if(you.inv.begin(), you.inv.end(), [&](const item_def& item) {
+        return item.base_type == OBJ_FOOD && item.sub_type == FOOD_CHUNK && !is_bad_food(item);
+    });
+    return it == you.inv.end() ? 0 : it->quantity;
+}
+
 /// Should the player automatically butcher the given item?
 static bool _should_autobutcher(const item_def &item)
 {
     return Options.auto_butcher >= you.hunger_state
            && item.base_type == OBJ_CORPSES
-           && !is_inedible(item) && !is_bad_food(item);
+           && !is_inedible(item) && !is_bad_food(item) 
+           && (Options.auto_butcher_max_chunks == 0
```

Comment:
I'd rather create a local named `max_chunks` or something holding this value, so we don't have to refer to this really long option name twice.